tools/alias_collision/check_alias_collision.py

```diff
@@ -0,0 +1,148 @@
+"""Check for alias collisions within the codebase"""
+
+from argparse import ArgumentDefaultsHelpFormatter, ArgumentParser
+from pathlib import Path
+from dataclasses import dataclass
+from typing import List, Dict
+import itertools
+import re
+
+
+ERROR_MESSAGE_TEMPLATE = (
+    "Alias `%s` defined in `%s` already exists as alias `%s` in `%s`."
+)
+
+# TODO: We want that list to be empty
+KNOWN_COLLISIONS = [
```

Comment:
We can reduce this list in the future. For now, we prevent new duplicates from being introduced

Reply:
What about comparing the duplicates from `master` (in the target commit, or latest, we can think about it) and the ones from the PR? I think it's better than having a list.
Imagine if we add an alias called `bcubc` besides the two that are already in the project.

Reply:
How about explicitly listing the full `Alias` instances? IMO the benefit of having such a list is that you know exactly how far you are. Comparing against master can be a moving target.

Reply:
You could export them as JSON or something like that and then parse the answer.

Reply:
Done